sklearn/_engine/base.py

```diff
@@ -112,3 +112,37 @@ def get_engine_classes(engine_name, default, verbose=False):
                 f"trying engine {engine_class.__module__}.{engine_class.__qualname__} ."
             )
         yield provider, engine_class
+
+
+def convert_attributes(method):
```

Comment:
Note for future iteration: Could method decorated by `convert_attributes` be called concurrently? If it is the case, should we add some handling to make converting attributes safe?